content/topics/jasmine-spies/_index.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Jasmine Spies
+ready: True
+---
+
+### Spying on JavaScirpt Methods using Jasmine
+
+Jasmine is made for unit testing. Unit tests are supposed to test only one component of your application. A component can be a function, an object, a module, basically everything self-contained that acts like a black box to the outside world. You usually want to avoid that your unit tests fail because another component failed. That’s why you want to test your components in isolation as much as possible.
+
+For example, you may not want to send data to another server in your unit tests or you don’t want to manipulate a page’s DOM. But you want to make sure that the components which are responsible for these tasks get called correctly. How do we do this?
```

Comment:
I assume they already know what jasmine is or what it does, just post a link to the jasmine topic and live out the description.

Reply:
Cool, but i just wanted to make it clear that components should be tested in isolation, i just feel like that has to be explained some how. would it be ok, if i rephrase the paragraph? @SweetBeard30

Reply:
It's a topic on advanced Jasmine usage,  it won't be novices reading this and they should already been using Jasmine if they've reached this point.

Reply:
Jasmine spies is an advace topic, anyone who reaches this point has probably been using jasmine already and are not new to it.

Reply:
@SweetBeard30 so i must rephrase the paragraph. is that what you are suggesting?

Reply:
@Sbuda32 remove it.

Reply:
Eish, sorry apparently I left a comment not a review. lol. https://github.com/Umuzi-org/tech-department/pull/323#pullrequestreview-323069927